src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/ClientManager.java

```diff
@@ -0,0 +1,122 @@
+package com.snowflake.kafka.connector.internal.ingestsdk;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.internal.LoggerHandler;
+import com.snowflake.kafka.connector.internal.SnowflakeErrors;
+import com.snowflake.kafka.connector.internal.streaming.StreamingUtils;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+public class ClientManager {
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+
+  private LoggerHandler LOGGER;
+  private int streamingIngestClientCount;
+  private SnowflakeStreamingIngestClient streamingIngestClient;
+
+  /** This is a wrapper to help manage the streaming ingest client */
+  protected ClientManager() {
+    LOGGER = new LoggerHandler(this.getClass().getName());
+    this.streamingIngestClientCount = 0;
+  }
+
+  // ONLY FOR TESTING - use this to inject client
+  @VisibleForTesting
+  public ClientManager(SnowflakeStreamingIngestClient client) {
+    this();
+    this.streamingIngestClient = client;
+  }
+
+  /**
+   * Calls the ingest sdk to create the streaming client
```

Comment:
Is it a single client or multiple clients that this client manager keeps track of? Should it be instead `a streaming client`?

Reply:
currently only one client, but i'm making the change to have multiple clients. thanks for bring this up